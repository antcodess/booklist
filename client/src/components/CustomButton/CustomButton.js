import React from 'react';

import './CustomButton.scss';

const CustomButton = ({ children, isGoogleSignIn, isRegularSignIn, ...otherProps }) => {
  return (
    <button 
    className={`custom-button ${isGoogleSignIn ? 'google-sign-in' : ''}${isRegularSignIn ? 'regular-sign-in' : ''}`} 
    {...otherProps}>{ children }</button>
  );
};

export default CustomButton;