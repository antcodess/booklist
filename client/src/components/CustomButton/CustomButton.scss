.regular-sign-in {
  margin-bottom: 5px;
  &:hover {
    background-color: black;
    opacity: 0.9;
  }
}

.google-sign-in {
  background-color: #0080DE;

  &:hover {
    opacity: 0.9;
  }
}

@media only screen and (min-width: 950px) {
  .regular-sign-in {
    width: 40%;
  }
}

@media only screen and (min-width: 1100px) {
  .regular-sign-in, .google-sign-in {
    width: 49%;
  }
}