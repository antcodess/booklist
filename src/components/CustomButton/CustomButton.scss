.regular-sign-in {
  margin-bottom: 5px;
}

.google-sign-in {
  background-color: #0080DE;

  &:hover {
    background-color: black;
  }
}