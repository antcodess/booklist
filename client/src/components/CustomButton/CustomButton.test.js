import React from 'react';
import { shallow } from 'enzyme';
import CustomButton from './CustomButton';

describe('CustomButton Component', () => {
  let wrapper;
  
  beforeEach(() => {

    const mockProps = {
      children: 'Add To Cart', 
      isGoogleSignIn: undefined, 
      isRegularSignIn: true
    };

    wrapper = shallow(<CustomButton { ...mockProps } />);
  });

  it('should render CustomButton Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should show regular button', () => {
    expect(wrapper.exists('.regular-sign-in')).toBe(true);
  });

  it('should be truthy if isRegularSignIn is true and falsy for isGoogleSignIn', () => {
    const mockProps = {
      children: 'Add To Cart', 
      isGoogleSignIn: undefined, 
      isRegularSignIn: true
    };
    wrapper = shallow(<CustomButton { ...mockProps } />);
    expect(mockProps.isRegularSignIn).toBeTruthy();
    expect(mockProps.isGoogleSignIn).toBeFalsy();
    expect(mockProps.isGoogleSignIn).toBeUndefined();
  });

  it('should be truthy if isGoogleSignIn is true and falsy for isRegularSignIn', () => {
    const mockProps = {
      children: 'Add To Cart', 
      isGoogleSignIn: true, 
      isRegularSignIn: undefined
    };
    wrapper = shallow(<CustomButton { ...mockProps } />);
    expect(mockProps.isGoogleSignIn).toBeTruthy();
    expect(mockProps.isRegularSignIn).toBeFalsy();
    expect(mockProps.isRegularSignIn).toBeUndefined();
  });
});