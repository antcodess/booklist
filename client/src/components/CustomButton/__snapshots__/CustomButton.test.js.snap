// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton Component should render CustomButton Component 1`] = `ShallowWrapper {}`;
